internal/manifests/targetallocator/deployment.go

```diff
@@ -44,7 +44,7 @@ func Deployment(params manifests.Params) (*appsv1.Deployment, error) {
 		Spec: appsv1.DeploymentSpec{
 			Replicas: params.OtelCol.Spec.TargetAllocator.Replicas,
 			Selector: &metav1.LabelSelector{
-				MatchLabels: labels,
+				MatchLabels: SelectorLabels(params.OtelCol),
```

Comment:
Isn't this immutable field? https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#label-selector-updates 

Can you confirm that this will not break during the upgrade?

Reply:
I'll test it. If there's a problem with that, then it's probably for the best to leave the selector as is.

Reply:
@pavolloffay we do have logic in our reconciler to just delete and recreate when it sees this immutable change, so it should just work (with a small blip for the reconciler logic)

Reply:
Deleting and recreating resources is a bit heavyhanded though, and I'm not sure if we want to do it just to have uniform selector labels between the collector and target allocator.

Reply:
I brought this label back. I don't think it makes the code substantially messier - check it our for yourselves.

Reply:
fair enough! Maybe lets keep it as it is then, and leave a comment explaining the divergence between the two.